src/snowflake/connector/connection.py

```diff
@@ -307,6 +314,18 @@ def __init__(self, **kwargs) -> None:
                 kwargs["application"] = "streamlit"
 
         self.converter = None
+        if config_file_path is not None:
+            # Change config file path and force update cache
+            CONFIG_PARSER.file_path = config_file_path
+            CONFIG_PARSER.read_config()
+        if connection_name is not None:
```

Comment:
When `config_file_path` and `connection_name` are not `None`, and the user also provide some kwargs, what will happen?

Reply:
Plus, will `SnowflakeConnection()` use the default connection name?

Reply:
> When config_file_path and connection_name are not None, and the user also provide some kwargs, what will happen?

All other kwargs will override what we load from env variables, or config file.

Reply:
> Plus, will `SnowflakeConnection()` use the default connection name?

I played around with the idea of having a default connection_name. I've decided against it for a few reasons:

* There's currently no way to see where a connection loaded its initial values from and which values came from overriding and which came from loading values.
* It could also lead to a forgotten setting put into the default connection and affecting code at run-time on a machine, so an identical script might execute very differently based on its environment.
* If this is necessary I think it's really easy for a user to use [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial) to achieve this behavior.